### PR TITLE
Add strong typing for API call params

### PR DESCRIPTION
**Pull request by Adlai-Holler**
Migrated all the existing `Methods` except `claim_search` which is big enough for its own PR.

### CODE REVIEW

File: Odysee/Utils/Extensions.swift [hunk 1]
```diff
@@ -107,3 +107,76 @@ extension Optional {
...
+        try c.encodeNil()
+    }
+}
+
+extension NSDictionary: Encodable {
+    public func encode(to encoder: Encoder) throws {
+        var c = encoder.container(keyedBy: CodingKeys.self)
+        for (k, v) in self {
+            guard let k = k as? String else {
+                throw EncodingError.invalidValue(k, .init(codingPath: encoder.codingPath,
```
Comment (akinwale) [inline Odysee/Utils/Extensions.swift:131]: Getting a compile error here and on a couple more lines. This is probably due to the lower Swift / XCode version which I have installed.

/Users/akinwale/Dev/iOS/Odysee/Odysee/Utils/Extensions.swift:131:50: Variable declared in 'guard' condition is not usable in its body

File: Odysee/Utils/Extensions.swift [hunk 2]
```diff
@@ -107,3 +107,76 @@ extension Optional {
...
+                                                          debugDescription: "NSDictionary key"))
+            }
+            let nestedEncoder = c.superEncoder(forKey: .init(stringValue: k))
+            guard let v = v as? Encodable else {
+                throw EncodingError.invalidValue(v, .init(codingPath: nestedEncoder.codingPath,
```
Comment (akinwale) [inline Odysee/Utils/Extensions.swift:136]: /Users/akinwale/Dev/iOS/Odysee/Odysee/Utils/Extensions.swift:136:50: Variable declared in 'guard' condition is not usable in its body

File: Odysee/Utils/Extensions.swift [hunk 3]
```diff
@@ -107,3 +107,76 @@ extension Optional {
...
+    }
+}
+
+extension NSArray: Encodable {
+    public func encode(to encoder: Encoder) throws {
+        var c = encoder.unkeyedContainer()
+        for v in self {
+            let nestedEncoder = c.superEncoder()
+            guard let v = v as? Encodable else {
+                throw EncodingError.invalidValue(v, .init(codingPath: nestedEncoder.codingPath,
```
Comment (akinwale) [inline Odysee/Utils/Extensions.swift:158]: /Users/akinwale/Dev/iOS/Odysee/Odysee/Utils/Extensions.swift:158:50: Variable declared in 'guard' condition is not usable in its body

**Review by akinwale — CHANGES_REQUESTED** [review]

File: Odysee/Utils/Extensions.swift (hunk 1 above)
Comment (Adlai-Holler) [reply Odysee/Utils/Extensions.swift:131]: Updated

**Review by akinwale — APPROVED** [review]
Looks good. Thanks.

**Pull request merged**